### ISSUE TITLE
android.database.CursorIndexOutOfBoundsException Index 0 requested, with a size of 0

### ISSUE THREAD

**Issue by Lir10** [coomar2841/android-multipicker-library#130]
**Trace:**

```
Fatal Exception: java.lang.RuntimeException
Failure delivering result ResultInfo{who=null, request=8666, result=-1, data=Intent { dat=content://com.android.contacts/contacts/lookup/3933i100005016570475/28 flg=0x1 }} to activity {com.test.test/com.test.test.CombineScreen}: android.database.CursorIndexOutOfBoundsException: Index 0 requested, with a size of 0
android.app.ActivityThread.deliverResults (ActivityThread.java:4920)
android.app.ActivityThread.handleSendResult (ActivityThread.java:4963)
android.app.ActivityThread.access$1600 (ActivityThread.java:221)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1848)
android.os.Handler.dispatchMessage (Handler.java:102)
android.os.Looper.loop (Looper.java:158)
android.app.ActivityThread.main (ActivityThread.java:7224)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1230)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1120)
Caused by android.database.CursorIndexOutOfBoundsException
Index 0 requested, with a size of 0
android.database.AbstractCursor.checkPosition (AbstractCursor.java:460)
android.database.AbstractWindowedCursor.checkPosition (AbstractWindowedCursor.java:136)
android.database.AbstractWindowedCursor.getInt (AbstractWindowedCursor.java:68)
android.database.CursorWrapper.getInt (CursorWrapper.java:122)
com.kbeanie.multipicker.api.ContactPicker.getRawContactId (ContactPicker.java:123)
com.kbeanie.multipicker.api.ContactPicker.queryForContact (ContactPicker.java:148)
com.kbeanie.multipicker.api.ContactPicker.submit (ContactPicker.java:110)
com.mynavy.intercomapp.CombineScreen.onActivityResult (CombineScreen.java:125)
android.app.Activity.dispatchActivityResult (Activity.java:7137)
android.app.ActivityThread.deliverResults (ActivityThread.java:4916)
android.app.ActivityThread.handleSendResult (ActivityThread.java:4963)
android.app.ActivityThread.access$1600 (ActivityThread.java:221)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1848)
android.os.Handler.dispatchMessage (Handler.java:102)
android.os.Looper.loop (Looper.java:158)
android.app.ActivityThread.main (ActivityThread.java:7224)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1230)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1120)
```

**Solution:**

in this line :         https://github.com/coomar2841/android-multipicker-library/blob/f9afe1cc6414a15a78acbc01d5962fa36b5cef6d/multipicker/src/main/java/com/kbeanie/multipicker/api/ContactPicker.java#L122

Change this:

```
 if (cursor.moveToFirst()) {
            rawContactId = cursor.getInt(cursor.getColumnIndexOrThrow(ContactsContract.RawContacts._ID));
        }
cursor.close();
```

to this:

 ```
if (cursor != null  && cursor.getCount()>0) {
           cursor.moveToFirst();
            rawContactId = cursor.getInt(cursor.getColumnIndexOrThrow(ContactsContract.RawContacts._ID));
cursor.close();
        }
```
Also i have noticed that sometimes if you fetch again it does return contacts
So maybe you can do:

 ```
if (cursor != null  && cursor.getCount()>0) {
           cursor.moveToFirst();
            rawContactId = cursor.getInt(cursor.getColumnIndexOrThrow(ContactsContract.RawContacts._ID));
cursor.close();
        }
else {
refetch(); // try again
}
```

**Comment by coomar2841** [comment]
Fixed. Will be released with version 1.5

**Comment by coomar2841** [comment]
Released with v1.5
